_scalingo_api/log_drains.md

```diff
@@ -0,0 +1,148 @@
+
+{:.table}
+| Logs management service   | type        | token | host | port | drain_region | url | will format urls:                                    |
+| ------------------------- | ------------| ----- | ---- | ---- | ------------ | --- | ---------------------------------------------------- |
+| Datadog                   | datadog     | ✓     |      |      | ✓            |     | datadog://`TOKEN`?region=`DRAIN_REGION`              |
+| OVH hosted Graylog        | ovh-graylog | ✓     | ✓    |      |              |     | ovh://:`TOKEN`@`HOST`:6514                           |
+| Logentries                | logentries  | ✓     |      |      |              |     | https://webhook.logentries.com/noformat/logs/`TOKEN` |
+| Papertrail                | papertrail  |       | ✓    | ✓    |              |     | tcp+tls://`HOST`:`PORT`                              |
+| Syslog                    | syslog      |       | ✓    | ✓    |              |     | tcp+tls://`HOST`:`PORT`                              |
+| Self-hosted ELK stack     | elk         |       |      |      |              | ✓   | `URL`                                                |
+
+--- row ---
+
+**log drain attributes**
+
+{:.table}
+| field               | type    | description                                                      |
+| ------------------- | ------  | ---------------------------------------------------------------- |
+| type                | string  | Communication protocol with the logs management service          |
+| token               | string  | Token used by logs management service for authentication         |
+| host                | string  | Host of logs management service                                  |
+| port                | string  | Port of logs management service                                  |
+| drain-region        | string  | Region used by logs management service to identify their servers |
+| url                 | string  | URL of self hosted ELK                                           |
+
+
+Note: `type` is mandatory.
+
+||| col |||
+
+--- row ---
+
+## List log drains of an app
+
+--- row ---
+
+`GET https://$SCALINGO_API_URL/v1/apps/:app/log_drains`
+
+||| col |||
+
+Example request
+
+```shell
+curl -H 'Accept: application/json' -H 'Content-Type: application/json' -u ":$AUTH_TOKEN" \
+  -X GET https://$SCALINGO_API_URL/v1/apps/example-app/log_drains
+```
+
+Returns 200 OK
+
+```json
+[
+    {
+        "app_id": "5af97be7ff688d0001228ffb",
+        "url": "ovh://:id@tag1.logs.ovh.com:6514"
+    },
+    {
+        "app_id": "5af97be7ff688d0001228ffb",
+        "url": "syslog://:id@custom.logs.com:port"
+    }
+]
+```
+
+--- row ---
+
+## Create a new log drain
+
+--- row ---
+
+`POST https://$SCALINGO_API_URL/v1/apps/:app/log_drains`
+
+### Parameters
+
+* `type`: can be any supported log management services type (e.g. papertrail, datadog...)
+* `token`: used by certain vendor for authentication
```

Comment:
```suggestion
* `token`: used by certain vendor for authentication (optional)
```

Can you add `(optional)` to the end of each optional items?

Reply:
I don't know if we need to add `optional` for the url parameter, because it is required in case of self hosted ELK as explained. What do you think ?

Reply:
But `url` is not present in case of other types of log drains, is it? That's why I would make it optional. I.e. don't include `url` for all types.

Reply:
Okay it was just to be sure :+1: